docker/ci/build.ps1

```diff
@@ -19,27 +19,45 @@ function Get-ContainerVersion
   (git describe) -replace '-.*', ''
 }
 
-# installs gems for build and test and grabs base images
-function Invoke-ContainerBuildSetup
+function Lint-Dockerfile($Path)
 {
-  Push-Location (Get-CurrentDirectory)
-  bundle install --path '.bundle/gems'
-  bundle exec puppet-docker update-base-images ubuntu:16.04
-  Pop-Location
+  hadolint --ignore DL3008 --ignore DL3018 --ignore DL4000 --ignore DL4001 $Path
 }
 
-function Build-Container($Name, $Repository = '127.0.0.1', $Version = (Get-ContainerVersion))
+function Build-Container(
+  $Name,
+  $Namespace = 'puppet',
+  $Version = (Get-ContainerVersion),
+  $Vcs_ref = $(git rev-parse HEAD))
 {
   Push-Location (Join-Path (Get-CurrentDirectory) '..')
-  bundle exec puppet-docker local-lint $Name
-  bundle exec puppet-docker build $Name --no-cache --repository $Repository --version $Version --no-latest --build-arg namespace=$Repository
+
+  $build_date = (Get-Date).ToUniversalTime().ToString('o')
+  $docker_args = @(
+    '--pull',
+    '--build-arg', "vcs_ref=$Vcs_ref",
+    '--build-arg', "build_date=$build_date",
+    '--build-arg', "version=$Version",
+    '--file', "$Name/Dockerfile",
+    '--tag', "$Namespace/$Name:$Version",
```

Comment:
I noticed we're not tagging as latest here like we are in puppetdb...

Turns out the Makefile actually does the tagging as latest via `docker tag $name:$version $name:latest` (in both PuppetDB and PuppetServer) so I don't think we need to worry about "latest" at this level.